docs/mobile/exercises/README.md

```diff
@@ -0,0 +1,284 @@
+relevant exercise type. Don't change this unless you absolutely need to for an
+exercise to make sense.
+
+## code
+
+```yaml
+  code: "print(42)"
+```
+
+This is the code! Anything you here should be syntactically correct, executable
+code. The format of the code block can vary slightly for different exercise
+types, for example reorder and tap exercises. See the documentation for those
+exercise types to get the details.
+
+## output
+
+```yaml
+  output: "42"
+```
+
+This is the output generated by the code in a code block. That means anything
+that would get printed to the console in an interactive Python or R terminal
+session.
+
+## distractors
+
+```yaml
+  distractor_output:
+    - option: "4"
+      feedback: "`print()` will output all of whatever is between the parentheses, not just the first part."
+    - option: "2"
+      feedback: "`print()` will output all of whatever is between the parentheses, not just the last part."
+```
+
+The rendered exercise type depends largely on the distractors block. In the
+example above, the use of the `distractor_output` field indicates this exercise
+will be a select output exercise.
+
+Other distractor types will result in different exercise types:
+
+- distractor_output -> [Select Output](select-output.md)
+- distractor_code   -> [Select Code](select-code.md)
+- distractor_blanks -> [Tap](tap.md)
+- distractor_text   -> [Multiple Choice](multiple-choice.md)
+- distractor_table  -> [Select Table](select-table.md)
+
+The only exercise type that does not require a distractor field is the
+[Reorder](reorder.md) .
+
+Regardless of the exercise type, the distractor field is always a list of
+distractors (AKA wrong answers). Each individual distractor in the list has two
+elements: `option`, the incorrect answer itself, and `feedback`, the feedback
+message that students will see if they incorrectly choose that distractor.
+
+# Other available blocks
+
+## table
+
+```yaml
+table:
+  data: |-
+    name     ,birthdate
+    50 Cent  ,1975-07-06
+    Aaliyah  ,1979-01-16
+    Aaron Yoo,1979-05-12
+  message: "Showing 3 out of 537 rows"
+```
+
+You can opt to use a `table` block instead of an `output` block, for example
+when displaying the output of a SQL query, which is more appropriate as a
+table. The table field has two subfields. First is `data`, which contains the
+actual table data as comma separated values. They needn't be aligned, but they
+can be and sometimes this improves readability.  The second is `message`, which
+contains a short string of text that is displayed in the footer of the
+table. The most common use for the table message is to indicate row truncation,
+because tables should not be displayed with more than 5 rows as a general rule.
+
+## image
+
+```yaml
+image: my-cool-plot
+```
+
+You can put image assets into your exercises! In this example, `my-cool-plot` is
+a reference to an image asset specified in the [course
+manifest](../repo-structure.md).
```

Comment:
Not necessary anymore, we should sync on this though as it is a bad idea to go and remove them from old courses.

Reply:
Nice - yes lets sync on what this looks like now